flusher.go

```diff
@@ -24,6 +24,9 @@ import (
 
 // Flush collects sampler's metrics and passes them to sinks.
 func (s *Server) Flush(ctx context.Context) {
+	s.flushMutex.Lock()
```

Comment:
If a sync is slow there's a change in behavior, is that correct?

E.g. let's say a sync flushes every 10s, but takes 11s. Before:
Syncs would overlap. E.g. we'd have 2 concurrent slow syncs running ~every 10s (1s of 2, 9s of 1)
The external systems would get data 11s delayed, but still every 10s.

After:
Syncs would pile up to be slower. E.g. we'd always only have 1 running, but we'd not be flushing every 10s but instead every 11s. We'd miss 1 in 10 flushes.
The external systems would get data every 11s.